Sources/TuistKit/Commands/InitCommand.swift

```diff
@@ -37,6 +37,27 @@ enum InitCommandError: FatalError, Equatable {
     }
 }
 
+enum Template: String {
```

Comment:
Templating would be a fantastic addition to generation of tuist projects ... how about instead of pre-computed variants we consider using a system similar to Xcode templates?

e.g.

```
SampleApp.xctemplate$ ll
drwxr-xr-x oat01 BSKYB\Domain Users 192 B  Wed Aug  7 14:58:06 2019   .
drwxr-xr-x oat01 BSKYB\Domain Users 160 B  Wed Aug  7 14:58:06 2019   ..
.rw-r--r-- oat01 BSKYB\Domain Users   2 KB Wed Aug  7 14:58:06 2019   AppDelegate.swift
.rw-r--r-- oat01 BSKYB\Domain Users 1.6 KB Wed Aug  7 14:58:06 2019   Main.storyboard
.rw-r--r-- oat01 BSKYB\Domain Users 1.2 KB Wed Aug  7 14:58:06 2019   TemplateInfo.plist
.rw-r--r-- oat01 BSKYB\Domain Users 274 B  Wed Aug  7 14:58:06 2019   ViewController.swift
```

**ViewController.swift**
```swift
//___FILEHEADER___

import UIKit
import ___VARIABLE_productName:identifier___

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

}
```

With this, tuist is agnostic of the type of template you want to define, allowing you to define as many as you want for any sort of application or framework you so desire. We can use a templating system such as [Stencil](https://github.com/stencilproject/Stencil)

Reply:
The initial idea was to get out of the gate SwiftUI first and then try to tackle a more generic approach for creating new frameworks, etc. in already initialized project.

But I suppose, as there seems to be quite high enthusiasm about it, that we should first figure out if `init --template swiftUI` can be replaced by something more ubiquitous, too (and thus probably should not be implemented at all).

I think what you are proposing is quite similar to `scaffold` feature from @pepibumur's [comment](https://github.com/tuist/tuist/issues/450#issuecomment-563026514). Stencil would be a great solution for this since it's usually a go-to in Swift community for generating code.

For this to work for `tuist init`, we'd need to first generate those templates and then, more-or-less, run the `scaffold` feature to make this work without hardcoding the templates here. This should work, though, if someone else confirms my assumption, I will probably close this PR and we can continue working on `scaffold` feature somewhere else in the shape as you and @pepibumur have laid out.

The other solution would be to have system-wide templates, similar to Xcode templates - but those become a bit hard to manage, so I think keeping everything in the local directory of the project is a better approach here.

Reply:
> For this to work for tuist init, we'd need to first generate those templates and then, more-or-less, run the scaffold feature to make this work without hardcoding the templates here. This should work, though, if someone else confirms my assumption, I will probably close this PR and we can continue working on scaffold feature somewhere else in the shape as you and @pepibumur have laid out.

I'd implement a templates engine as @ollieatkinson suggested too. I think we can start by supporting built-in templates, and then extend it to allow users to define their own templates, for example under `Tuist/Templates/...`. This is a very powerful idea because that means that at any directory, you'll be able to create new projects by just running a command.